### ISSUE TITLE
method does not override or implement a method from a supertype on v0.47

### ISSUE THREAD

**Issue by thevishnupradeep** [invertase/react-native-firebase#300]
I am getting an error "method does not override or implement a method from a supertype..." on every RNFirebase module in Android when using 0.47. The same error disappears when downgraded to 0.46.4 (there is another error, but I don't think it is related to RNFirebase).

Could someone help? 

This kind of error is showing up for all packages.
`
/Users/vishnu/Documents/Prioritize/node_modules/react-native-firebase/android/src/main/java/io/invertase/firebase/admob/RNFirebaseAdMobPackage.java
Error:(39, 3) error: method does not override or implement a method from a supertype
`

**Comment by hofuchi** [comment]
Here's some compiler output for `react-native run-android`. Same errors when building release version

```
:react-native-firebase:incrementalReleaseJavaCompilationSafeguard UP-TO-DATE
:react-native-firebase:compileReleaseJavaWithJavac
:react-native-firebase:compileReleaseJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
/home/gscanlon/Documents/reversedictionary/node_modules/react-native-firebase/android/src/main/java/io/invertase/firebase/messaging/RNFirebaseMessagingPackage.java:38: error: method does not override or implement a method from a supertype
  @Override
  ^
/home/gscanlon/Documents/reversedictionary/node_modules/react-native-firebase/android/src/main/java/io/invertase/firebase/storage/RNFirebaseStoragePackage.java:38: error: method does not override or implement a method from a supertype
  @Override
  ^
/home/gscanlon/Documents/reversedictionary/node_modules/react-native-firebase/android/src/main/java/io/invertase/firebase/database/RNFirebaseDatabasePackage.java:38: error: method does not override or implement a method from a supertype
  @Override
  ^
/home/gscanlon/Documents/reversedictionary/node_modules/react-native-firebase/android/src/main/java/io/invertase/firebase/crash/RNFirebaseCrashPackage.java:38: error: method does not override or implement a method from a supertype
  @Override
  ^
/home/gscanlon/Documents/reversedictionary/node_modules/react-native-firebase/android/src/main/java/io/invertase/firebase/config/RNFirebaseRemoteConfigPackage.java:38: error: method does not override or implement a method from a supertype
  @Override
  ^
/home/gscanlon/Documents/reversedictionary/node_modules/react-native-firebase/android/src/main/java/io/invertase/firebase/auth/RNFirebaseAuthPackage.java:38: error: method does not override or implement a method from a supertype
  @Override
  ^
/home/gscanlon/Documents/reversedictionary/node_modules/react-native-firebase/android/src/main/java/io/invertase/firebase/perf/RNFirebasePerformancePackage.java:38: error: method does not override or implement a method from a supertype
  @Override
  ^
/home/gscanlon/Documents/reversedictionary/node_modules/react-native-firebase/android/src/main/java/io/invertase/firebase/admob/RNFirebaseAdMobPackage.java:39: error: method does not override or implement a method from a supertype
  @Override
  ^
/home/gscanlon/Documents/reversedictionary/node_modules/react-native-firebase/android/src/main/java/io/invertase/firebase/analytics/RNFirebaseAnalyticsPackage.java:38: error: method does not override or implement a method from a supertype
  @Override
  ^
/home/gscanlon/Documents/reversedictionary/node_modules/react-native-firebase/android/src/main/java/io/invertase/firebase/RNFirebasePackage.java:41: error: method does not override or implement a method from a supertype
  @Override
  ^
Note: /home/gscanlon/Documents/reversedictionary/node_modules/react-native-firebase/android/src/main/java/io/invertase/firebase/auth/RNFirebaseAuth.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
10 errors
:react-native-firebase:compileReleaseJavaWithJavac FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-firebase:compileReleaseJavaWithJavac'.
> Compilation failed; see the compiler error output for details.
```

I got the same error when upgrading react-native to `0.47.1`

```
"dependencies": {
    "react": "16.0.0-alpha.12",
    "react-native": "^0.47.1",
    "react-native-firebase": "^2.0.5",
  },
```

**Comment by Ehesp** [comment]
This should fix the breaking change, https://github.com/invertase/react-native-firebase/pull/299, can you try that PR for us? Need to double check it works backwards before we merge but that should help.

**Comment by hofuchi** [comment]
Sure I'll try it

**Comment by evgeni-tsn** [comment]
Hey @Ehesp can we get this merge any time soon? I just started my new project and I want all to be clear in setup process. Thanks!

**Comment by chrisbianca** [comment]
Hey guys, I have just released v2.1.0 which supported RN 0.47 and earlier versions.  Thanks for reporting the issue.
